packages/core/content-releases/server/src/services/release.ts

```diff
@@ -230,6 +231,34 @@ const createReleaseService = ({ strapi }: { strapi: LoadedStrapi }) => ({
 
     return contentTypesMeta;
   },
+  async delete(releaseId: DeleteRelease.Request['params']['id']) {
+    const release = await strapi.entityService.findOne(RELEASE_MODEL_UID, releaseId);
+
+    if (!release) {
+      throw new errors.NotFoundError(`No release found for id ${releaseId}`);
+    }
+
+    if (release.releasedAt) {
+      throw new errors.ValidationError('Release already published');
+    }
+
+    const deletedRelease = (await strapi.entityService.delete(RELEASE_MODEL_UID, releaseId, {
+      populate: { actions: { fields: ['id'] } },
+    })) as unknown as Release;
+
+    // We delete each action related to the release
+    if (deletedRelease.actions) {
+      await strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
+        where: {
+          id: {
+            $in: deletedRelease.actions.map(({ id }) => id),
+          },
+        },
+      });
+    }
```

Comment:
So no matter what we delete the release right? If something goes wrong deleting the actions then what will happen? An error throws in the deleteMany, the release is already deleted, and we return an error? Seems a bit misleading in that case.

I'm wondering if the order of operations matters here. delete the actions, when that's a success delete the release? 🤔

Reply:
I think the order you proposed is right, we should delete the actions firsts.

I also added a transaction for this, because in my opinion it's the perfect use case: We want to delete the release and all its actions only if we can delete all without problems 🤔